### PR TITLE
Fix logits_all parameter being ignored

### PR DESCRIPTION
**Pull request by crasm**
`batch.logits` is always true since the new batch API mallocs it. This made the other control flows dead code.

I am unsure what the last else statement was intended to do, and removed it because it has been dead code for a while anyway.

### CODE REVIEW

**Comment by cebtenzzre** [comment]
> `batch.logits` is always true since the new batch API mallocs it.

Not true. llama_eval and llama_batch_get_one still exist and set it to nullptr.
